### PR TITLE
Copy Wickets way of dealing with metadata. Also copy their JavaDoc :-)

### PR DESCRIPTION
**Pull request by matozoid**
Fixes issue #456 

I'll ask what the Wicket guys think of this :D

### CODE REVIEW

File: javaparser-testing/src/test/java/com/github/javaparser/ast/MetaDataKeyTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        node.setMetaData(ABC, "Hurray!");
+        node.setMetaData(LISTY, Arrays.asList("a", "b"));
+        node.setMetaData(ABC, "w00t");
+
+        assertThat(node.getMetaData(ABC)).contains("w00t");
+        assertThat(node.getMetaData(LISTY)).containsExactly("a", "b");
+        assertThat(node.getMetaData(DING)).isNull();
+    }
+
+}
```
Comment (ftomassetti) [inline javaparser-testing/src/test/java/com/github/javaparser/ast/MetaDataKeyTest.java]: Newline at the end of the file?

Comment (matozoid) [reply javaparser-testing/src/test/java/com/github/javaparser/ast/MetaDataKeyTest.java]: Newline at the end of the file...

**Comment by matozoid** [comment]
I think it looks very pretty. Thanks for the hint, @ptitjes !

**Comment by martin-g** [comment]
Hi,

I've been notified that you needed to copy some code from Apache Wicket for your needs.
Since Apache Wicket is ASL2 licenced you can do whatever you want with the source code without asking anyone!
If you want you can mention somewhere that your project has copied code from other projects. But this is not mandatory at all!

Good luck!

**Comment by ptitjes** [comment]
« Licenced to steal » ? :)

And BTW, I wasn't suggesting to copy but rather get inspired... The pattern in itself was well-known before Wicket makes use of it, as using an anonymous inner class is the only way to retrieve the type parameter at runtime. (Web-search for that.)

That said, IMHO the term 'data' would be a better fit than 'mata-data' here...

**Comment by matozoid** [comment]
I understand, the decision to copy (most) was completely mine :) I especially like the Javadoc. Maybe I should take Jonathan Locke's name out though!

Will change to data.

**Comment by ptitjes** [comment]
Or 'userData', or something else that makes it clear it is intended for the only use of the user, without reading the docs...

**Pull request merged**